cmd/cluster-init/main.go

```diff
@@ -0,0 +1,109 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/sirupsen/logrus"
+)
+
+type options struct {
+	clusterName string
+	releaseRepo string
+}
+
+func (o options) String() string {
+	return fmt.Sprintf("%#v", o)
+}
+
+func parseOptions() (options, error) {
+	var o options
+	fs := flag.NewFlagSet(os.Args[0], flag.ExitOnError)
+	fs.StringVar(&o.clusterName, "cluster-name", "", "The name of the new cluster.")
+	fs.StringVar(&o.releaseRepo, "release-repo", "", "Path to the root of the openshift/release repository.")
+
+	if err := fs.Parse(os.Args[1:]); err != nil {
+		return options{}, err
+	}
+	return o, nil
+}
+
+func validateOptions(o options) []error {
+	var errs []error
+	if o.clusterName == "" {
+		errs = append(errs, fmt.Errorf("--cluster-name must be provided"))
+	} else if strings.ContainsAny(o.clusterName, " \t\n") {
+		errs = append(errs, fmt.Errorf("--cluster-name must not contain whitespace"))
+	}
+	if o.releaseRepo == "" {
+		errs = append(errs, fmt.Errorf("--release-repo must be provided"))
+	}
+	if o.clusterName != "" {
+		buildDir := buildFarmDirFor(o.releaseRepo, o.clusterName)
+		if _, err := os.Stat(buildDir); !os.IsNotExist(err) {
+			errs = append(errs, fmt.Errorf("build farm directory: %s already exists", o.clusterName))
+		}
+	}
+	return errs
+}
+
+func main() {
+	o, err := parseOptions()
+	if err != nil {
+		logrus.WithError(err).Fatal("cannot parse args: ", os.Args[1:])
+	}
+	validationErrors := validateOptions(o)
+	if len(validationErrors) > 0 {
+		logrus.Fatalf("validation errors: %v", validationErrors)
+	}
+
+	// Each step in the process is allowed to fail independently so that the diffs for the others can still be generated
+	errorCount := 0
+	for _, step := range []func(options) error{
+		updateClustersReadme,
+		initClusterBuildFarmDir,
+		updateSanitizeProwJobs,
+	} {
+		if err := step(o); err != nil {
+			logrus.WithError(err)
+			errorCount++
+		}
+	}
+	if errorCount > 0 {
+		logrus.Fatalf("Due to the %d error(s) encountered a PR will not be generated. The resulting files can be PR'd manually", errorCount)
+	}
+}
+
+func updateClustersReadme(o options) error {
```

Comment:
Let's let the user provide a static input (using a flag to pass in text or a file) instead of anything interactive.

Reply:
As discussed we are just killing this feature for now. We can come back later and programmatically update the README if we think that actually provides value.